test/e2e/runner/docker.go

```diff
@@ -0,0 +1,219 @@
+// nolint: gosec
```

Comment:
If it's practical, I'd suggest putting this implementation into a non-main package and importing that. (Among other things that will make it a little nicer to write meta-tests against it).